### ISSUE TITLE
Add a default-selected property as selection fallback

### ISSUE THREAD

**Issue by TimvdLippe** [PolymerElements/iron-selector#116]
### Description
With the recent release of `carbon-route`, a current issue is a missing default routing fallback when for example a 404 is hit. The issue with the request of this feature: https://github.com/PolymerElements/carbon-route/issues/68

During the discussion, we discovered two possible solutions: creating a meta-element to track all route elements or a default fallback for selectors based on the route. After some thinking and prototyping, @rictic and me think that the simplest and most elegant solution is to allow a default/fallback for the selectors.

### Proposal
We propose a new property `defaultSelected` in `IronSelectableBehavior`. This property is a fallback when [`_valueToItem`](https://github.com/PolymerElements/iron-selector/blob/master/iron-selectable.html#L271) results in undefined and `this.attrForSelected` is set. In that case, [there was no match](https://github.com/PolymerElements/iron-selector/blob/master/iron-selectable.html#L277) for the specific route.

### Example use-case
A full use-case can be found here: https://gist.github.com/TimvdLippe/9bfebc20aec75dd36126b8162c310dff

The usage of `iron-pages`:
```html
<iron-pages attr-for-selected="data-route" selected="{{data.page}}" default-selected="404">
  <user-page data-route="users" user-info="{{subData}}"></user-page>
  <store-page data-route="store" store-info="{{subData}}"></store-page>
  <404-page data-route="404"></404-page>
</iron-pages>
```

### Compatibility
The implementation of this feature should be fairly easy and backwards-compatible.

One possible implementation could be:
```js
if (value == null) {
  return null;
}
var index = this._valueToIndex(value)
  || (this.defaultSelected && this._valueToIndex(this.defaultSelected));
return this.items[index];
```

**Comment by bicknellr** [comment]
This sounds like a good idea; want to send a PR?

**Comment by leodido** [comment]
+1

**Comment by TimvdLippe** [comment]
@bicknellr I submitted a PR in #117

**Comment by TimvdLippe** [comment]
For the record: the property was renamed to `fallbackSelection` and now available in 1.4.0
